src/zarr/core/group.py

```diff
@@ -2594,120 +2535,84 @@ def array(
         self,
         name: str,
         *,
-        shape: ChunkCoords,
-        dtype: npt.DTypeLike = "float64",
-        fill_value: Any | None = None,
+        shape: ShapeLike,
+        dtype: npt.DTypeLike,
+        chunks: ChunkCoords | Literal["auto"] = "auto",
+        shards: ChunkCoords | Literal["auto"] | None = None,
+        filters: Iterable[dict[str, JSON] | Codec] = (),
+        compressors: Iterable[dict[str, JSON] | Codec] = (),
```

Comment:
Could we use auto here?
```suggestion
        filters: Iterable[dict[str, JSON] | Codec] = "auto",
        compressors: Iterable[dict[str, JSON] | Codec] = "auto",
```